sql/core/src/main/scala/org/apache/spark/sql/sources/ddl.scala

```diff
@@ -83,10 +99,104 @@ private[sql] class DDLParser extends StandardTokenParsers with PackratParsers wi
   protected lazy val className: Parser[String] = repsep(ident, ".") ^^ { case s => s.mkString(".")}
 
   protected lazy val pair: Parser[(String, String)] = ident ~ stringLit ^^ { case k ~ v => (k,v) }
+
+  protected lazy val column: Parser[StructField] =
+  ( ident ~  ident ^^ { case name ~ typ =>
+      StructField(name, metastoreTypes.toDataType(typ))
```

Comment:
In order to handle also possible user definition, here we can make the type forced be lower case so __ toDataType__ can handle it correctly. Like:
```
StructField(name, metastoreTypes.toDataType(typ.toLowerCase)
```